iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/NetworkIndex.java

```diff
@@ -50,14 +53,54 @@ void checkAndAdd(Identifiable<?> obj) {
         all.add(obj);
     }
 
-    Identifiable get(String id) {
+    boolean addAlias(Identifiable<?> obj, String alias) {
+        if (objectsById.containsKey(alias)) {
+            if (objectsById.get(alias).equals(obj)) {
+                // Silently ignore affecting the objects id to its own aliases
+                return false;
+            }
+            Identifiable<?> aliasConflict = objectsById.get(alias);
+            String message = String.format("Object (%s) with alias '%s' cannot be created because alias already refers to object (%s) with ID '%s'",
+                    obj.getClass(),
+                    alias,
+                    aliasConflict.getClass(),
+                    aliasConflict.getId());
+            throw new PowsyblException(message);
+        } else if (idByAlias.containsKey(alias)) {
+            if (objectsById.get(idByAlias.get(alias)).equals(obj)) {
+                // Silently ignore affecting the same alias twice to an object
+                return false;
+            }
+            Identifiable<?> aliasConflict = objectsById.get(idByAlias.get(alias));
+            String message = String.format("Object (%s) with alias '%s' cannot be created because alias already refers to object (%s) with ID '%s'",
+                    obj.getClass(),
+                    alias,
+                    aliasConflict.getClass(),
+                    aliasConflict.getId());
+            throw new PowsyblException(message);
+        }
+        idByAlias.put(alias, obj.getId());
+        return true;
+    }
+
+    public <I extends Identifiable<I>> void removeAlias(Identifiable<?> obj, String alias) {
+        if (!idByAlias.containsKey(alias)) {
+            throw new PowsyblException(String.format("No alias '%s' found in the network", alias));
+        } else if (!idByAlias.get(alias).equals(obj.getId())) {
+            throw new PowsyblException(String.format("Alias '%s' do not correspond to object '%s'", alias, obj.getId()));
+        } else {
+            idByAlias.remove(alias);
+        }
+    }
+
+    Identifiable get(String idOrAlias) {
+        String id = idByAlias.getOrDefault(idOrAlias, idOrAlias);
```

Comment:
That means it's not possible to use the ID of an object as one of its alias. I know it doesn't make any sense, but if it's allowed by the code, does it work properly?

Reply:
In current implementation, ID is not in the alias list, but as it is the ID, it works to retrieve the object using the ID